### ISSUE TITLE
sbt 1.3.x can't resolve paths with spaces on Windows

### ISSUE THREAD

**Issue by kavedaa** [sbt/sbt#5433]
## steps

test-lib:
```
name := "test-lib"
organization := "no.vedaadata"
scalaVersion := "2.12.0"
```

test-proj:
```
name := "test-proj"
scalaVersion := "2.12.0"
libraryDependencies += "no.vedaadata" %% "test-lib" % "0.1.0-SNAPSHOT"
```

Tested with: sbt 1.2.8 and 1.3.8.

## problem

On 1.2.8 the dependency is resolved. On 1.3.8 it is not:

```
E:\prog\jvm\sbt\bugs\test-proj>sbt
[info] Loading settings for project global-plugins from dependency.sbt ...
[info] Loading global plugins from C:\Users\Knut Arne Vedaa\.sbt\1.0\plugins
[info] Loading project definition from E:\prog\jvm\sbt\bugs\test-proj\project
[info] Loading settings for project test-proj from test-proj.sbt ...
[info] Set current project to test-proj (in build file:/E:/prog/jvm/sbt/bugs/test-proj/)
[info] sbt server started at local:sbt-server-818527dc0e3d7a740335
sbt:test-proj> sbtVersion
[info] 1.2.8
sbt:test-proj> update
[info] Updating ...
[warn] Resolving a snapshot version. It's going to be slow unless you use `updateOptions := updateOptions.value.withLatestSnapshots(false)` options.
[info] Out of 1 candidates we found for no.vedaadata#test-lib_2.12;0.1.0-SNAPSHOT in local, we are choosing local.
[info] Done updating.
[success] Total time: 1 s, completed 15.feb.2020 13:16:19
sbt:test-proj> exit
[info] shutting down server

E:\prog\jvm\sbt\bugs\test-proj>sbt
[info] Loading settings for project global-plugins from dependency.sbt ...
[info] Loading global plugins from C:\Users\Knut Arne Vedaa\.sbt\1.0\plugins
[info] Loading project definition from E:\prog\jvm\sbt\bugs\test-proj\project
[info] Loading settings for project test-proj from test-proj.sbt ...
[info] Set current project to test-proj (in build file:/E:/prog/jvm/sbt/bugs/test-proj/)
[info] sbt server started at local:sbt-server-818527dc0e3d7a740335
sbt:test-proj> sbtVersion
[info] 1.3.8
sbt:test-proj> update
feb 15, 2020 1:16:38 PM lmcoursier.internal.shaded.coursier.cache.shaded.org.jline.utils.Log logr
WARNING: Unable to create a system terminal, creating a dumb terminal (enable debug logging for more information)
[info] Updating
[info] Resolved  dependencies
[warn]
[warn]  Note: Unresolved dependencies path:
[error] stack trace is suppressed; run last update for the full output
[error] (update) sbt.librarymanagement.ResolveException: Error downloading no.vedaadata:test-lib_2.12:0.1.0-SNAPSHOT
[error]   Not found
[error]   Not found
[error]   not found: C:\Users\Knut%20Arne%20Vedaa\.ivy2\local\no.vedaadata\test-lib_2.12\0.1.0-SNAPSHOT\ivys\ivy.xml
[error]   not found: https://repo1.maven.org/maven2/no/vedaadata/test-lib_2.12/0.1.0-SNAPSHOT/test-lib_2.12-0.1.0-SNAPSHOT.pom
```

**Comment by eed3si9n** [comment]
@kavedaa Thanks for the report. Are you sure if this has to do with whitespace?
Coursier uses different semantics for SNAPSHOTs - https://get-coursier.io/docs/ttl

**Comment by kavedaa** [comment]
Quite certain, as I can't reproduce it using an account with no spaces in the user's name. I also suspect it's related to the `%20` encoding:

`[error]   not found: C:\Users\Knut%20Arne%20Vedaa\.ivy2\local\no.vedaadata\test-lib_2.12\1.0\ivys\ivy.xml`

Normal spaces are used when publishing (which works with both sbt versions):

`[info]  published ivy to C:\Users\Knut Arne Vedaa\.ivy2\local\no.vedaadata\test-lib_2.12\1.0\ivys\ivy.xml`

Btw I also changed the version number from `-SNAPSHOT` to `1.0` and the problem is still present.

**Comment by eed3si9n** [comment]
ok. If it's likely a Coursier bug, then could you also report to coursier/coursier please?

**Comment by kavedaa** [comment]
https://github.com/coursier/coursier/issues/1581

**Comment by jastice** [comment]
This also reproduces on Mac (and presumably Linux), and also with non-SNAPSHOTs.

Error looks like this:
```
[error]   not found: /Users/jast/Library/Application%20Support/JetBrains/Toolbox/apps/IDEA-U/ch-1/203.5600.34/IntelliJ%20IDEA%202020.3%20EAP.app.plugins/Scala/repo/org.jetbrains.scala/sbt-idea-compiler-indices/scala_2.12/sbt_1.0/1.0.8/ivys/ivy.xml
```
The file exists, but the URL-encoding breaks the path.

One of the causes of https://youtrack.jetbrains.com/issue/SCL-18157 (sbt shell breaking in IntelliJ)

**Comment by alexarchambault** [comment]
FYI, this should be fixed in the upcoming lm-coursier release, thanks to https://github.com/coursier/coursier/pull/1916.

**Comment by eed3si9n** [comment]
Closing this since 1.4.5 was released with Coursier 2.0.7.
